src/code/game.c

```diff
@@ -392,20 +430,33 @@ void GameState_Init(GameState* gameState, GameStateFunc init, GraphicsContext* g
     gameState->destroy = NULL;
     gameState->running = 1;
     startTime = osGetTime();
+
+#ifdef OOT_DEBUG
     gameState->size = 0;
     gameState->init = NULL;
-    endTime = osGetTime();
+#else
+    gameState->init = NULL;
+    gameState->size = 0;
+#endif
 
-    // "game_set_next_game_null processing time %d us"
-    PRINTF("game_set_next_game_null 処理時間 %d us\n", OS_CYCLES_TO_USEC(endTime - startTime));
-    startTime = endTime;
-    GameAlloc_Init(&gameState->alloc);
+    {
+#ifndef OOT_DEBUG
+        s32 requiredScopeTemp;
+#endif
```

Comment:
for both this ifdef and the one above, is it possible to find code that matches both the retail and debug versions here?

Reply:
wasn't a better match found since? idr well

Reply:
Revo sent this yesterday: https://decomp.me/scratch/TBTcx if everyone's ok with this solution I'll use that instead

Reply:
Not sure, does that also match the debug version? Basically I think we should look for C code that matches both retail and debug without requiring any ifdefs

Reply:
doesn't seem to match the debug rom

Reply:
This matches both versions with no ifdefs:

```
    ...
    startTime = osGetTime();
    gameState->size = gameState->init = 0;

    {
        s32 requiredScopeTemp;
        endTime = osGetTime();
        ...
```

Reply:
Sorry, I mean the ifdef above too, so instead of

```
#ifdef OOT_DEBUG
    gameState->size = 0;
    gameState->init = NULL;
#else
    gameState->init = NULL;
    gameState->size = 0;
#endif
```

we can have
```
gameState->size = gameState->init = 0;
```

(GitHub would not let me turn this into a code suggestion)

Reply:
it's weird to not use ``NULL`` to initialise pointers imo

Reply:
Yeah I agree, but I still prefer it to the ifdef. Maybe there's another way to match it though

Reply:
this happens all the time unfortunately with things like bools as well. sometimes youve just gotta do it

Reply:
this is adding a warning though, is it ok? ``warning: assignment to ‘u32’ {aka ‘long unsigned int’} from ‘GameStateFunc’ {aka ‘void (*)(struct GameState *)’} makes integer from pointer without a cast [-Wint-conversion]``

Reply:
Technically you can avoid it in this case with
```
gameState->size = (u32)(gameState->init = NULL);
```
but to be honest I think I prefer the warning. I see other warnings in the build (including in game.c already) so I assume it's fine

Reply:
we only allow warnings if it reallllly warrants it (though they may not have been cleaned up in a long time)
it may be warranted in this case, but also MM doesnt need to do this. So something tells me there may be a better match

Reply:
MM just does 
```
    gameState->init = NULL;
    gameState->size = 0;
```
though, it doesn't need to match a debug version where the assignments are in a different order

Reply:
I went with ``gameState->size = gameState->init = 0;``, maybe we'll be able to get a better match later once more versions are done?